### PR TITLE
add 21 merge-two-sorted-lists

### PR DESCRIPTION
**Pull request by kobakobashu**
30m

### CODE REVIEW

**Comment by kobakobashu** [comment]
forget writing time and space complexity.
please wait

**Comment by kobakobashu** [comment]
Please check my code

File: 21.merge-two-sorted-lists.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#     def __init__(self, val=0, next=None):
+#         self.val = val
+#         self.next = next
+class Solution:
+    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
+        """
+        time: O(S) S: sum of the all node
+        space: O(S)
+        """
+        output = ListNode(0)
```
Comment (ryuji0123) [inline 21.merge-two-sorted-lists.py:19]:
```py
head = ListNode(0)
```

Comment (ryuji0123) [inline 21.merge-two-sorted-lists.py:19]:
```py
head = ListNode(0)
```

File: 21.merge-two-sorted-lists.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        output_add = output
```
Comment (ryuji0123) [inline 21.merge-two-sorted-lists.py:20]:
```py
cur_node = head
```

File: 21.merge-two-sorted-lists.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        while l1 and l2:
+            if l1.val <= l2.val:
+                output_add.next = ListNode(l1.val)
+                l1 = l1.next
+            else:
+                output_add.next = ListNode(l2.val)
+                l2 = l2.next
+            output_add = output_add.next
+        if not l1:
```
Comment (ryuji0123) [inline 21.merge-two-sorted-lists.py:29]:
```py
cur_node.next = l1 if not l2 else l2
```

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
common names for this kind of problems are:
```py
head, cur_node, prev_node, next_node, tail
```

**Comment by kobakobashu** [comment]
I fixed code.  Please check it

**Review by ryuji0123 — APPROVED** [review]
LGTM, but using head variable with dummy value is a little bit confusing. My answer is:

```py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        if not l1:
            return l2
        if not l2:
            return l1
        
        if l1.val <= l2.val:
            head = ListNode(l1.val)
            l1 = l1.next
        else:
            head = ListNode(l2.val)
            l2 = l2.next
        cur_node = head
        
        while l1 and l2:
            if l1.val <= l2.val:
                cur_node.next = ListNode(l1.val)
                l1 = l1.next
            else:
                cur_node.next = ListNode(l2.val)
                l2 = l2.next
            cur_node = cur_node.next
        
        cur_node.next = l1 if not l2 else l2
        
        return head
```

**Pull request merged**